src/ShallowRenderer.ts

```diff
@@ -63,9 +64,12 @@ export default class ShallowRenderer implements AbstractShallowRenderer {
     });
   }
 
-  simulateEvent(node: RSTNode, eventName: string, args: Object) {
+  simulateEvent(node: RSTNode, eventName: string, ...args: any[]) {
     withShallowRendering(() => {
-      this._mountRenderer.simulateEvent(node, eventName, args);
+      const handler = node.props[propFromEvent(eventName)];
+      if (handler) {
+        handler(...args);
```

Comment:
This looks like a significant change to how event simulation works with shallow rendering. Can you provide some more context here? This isn't going to cause a problem for the company I work for as we don't use shallow rendering, but it could potentially be a surprise for other users. If we do decide to do this, it ought to be marked as a breaking change.

Reply:
This change was just to bring our adapter into alignment with the React adapter. See my other comment for the source.

Yes, good call. We should release this as a breaking change. Should I retarget this PR to a "v5" branch we can release after gathering a couple of changes?

To be clear, my end goal here is to bring our enzyme adapter and react's enzyme adapter a bit closer in feature parity to make migrating easier.

Reply:
> Should I retarget this PR to a "v5" branch we can release after gathering a couple of changes?

Are there other breaking changes that you expect may be needed soon?

Reply:
I'm not sure yet. I'm still exploring/discovering what behaviors of the React 16 adapter our existing tests rely on, so it is possible. I imagine within a week or two I'll have a good grasp of what they are and if they are breaking or not.

Reply:
One possibility to avoid the need for a separate branch would be to add a feature flag in the adapter's configuration to control which behavior is used. The flag can be set to preserve the current behavior and then flipped in the next major release.